client/pages/matches.jsx

```diff
@@ -43,6 +44,29 @@ export default class Matches extends React.Component {
       });
   }
 
+  handleSave() {
+    fetch('/api/favorites', {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({
+        petId: Number(this.state.id),
+        userId: 1,
+        details: {
+          name: this.state.name,
+          photos: this.state.photos,
+          location: this.state.location,
+          age: this.state.age,
+          breed: this.state.breed
+        }
+      })
+    })
+      .then(response => response.json())
+      .catch(error => {
+        console.error('Error', error);
+      });
+    window.location.reload();
```

Comment:
if you want to fetch a new match, you can do that in your second `.then()` callback instead of reloading the whole universe.

Reply:
That definitely makes more sense! Will implement the revisions asap